### PR TITLE
Resolve mobile push state in join_bridge before pruning it

### PR DESCRIPTION
**Pull request by fblackburn1**
Why: join_bridge pruned the push state in its finally block, while the
completed-vs-cancelled decision lived in the AMI BridgeEnter bus
handler. Since join_bridge is what creates the bridge, the handler
structurally always ran after the prune: the completion path never
executed and every answered mobile-push call logged a
"cancel_push_mobile: no mobile push state" warning.

The decision now happens synchronously in join_bridge, which knows the
answering channel; the dead BridgeEnter handler is removed.

Side effect: when a pickup or another non-mobile device answers, the
push cancellation is now actually dispatched, so the mobile app
dismisses the incoming call instead of ringing until timeout.

---

> [!NOTE]
> **Medium Risk**
> Changes timing and ownership of mobile push lifecycle on answered calls—a user-visible telephony path—but the change aligns behavior with intended design and adds regression tests.
> 
> **Overview**
> **Fixes mobile push never completing on answer** because `join_bridge` always pruned push state in `finally` before the AMI `BridgeEnter` handler could decide complete vs cancel—so every answered call hit spurious cancel warnings and the mobile completion path never ran.
> 
> Push resolution now runs **inside `join_bridge`** (after answer, before bridge creation) via `_resolve_pending_push`, which inspects the answering channel with `_is_users_mobile_channel` (PJSIP user UUID + `mobility=mobile`) and calls `complete_pending_push_mobile` or `cancel_push_mobile`. The **`BridgeEnter` bus subscription and `has_a_registered_mobile_and_pending_push`** are removed; `DialEnd` comments note answered calls are handled by join.
> 
> **Behavioral fix:** when a desk phone, pickup, or other non-mobile leg answers, **cancel push is actually sent** so the mobile app stops ringing instead of timing out.
> 
> Adds an integration test for join-time cancel and broad unit coverage for resolution edge cases; drops obsolete `BridgeEnter` consumer tests.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit add69ce7b0a7f6c837af3ca3820995fcbb3c7c24. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: wazo_calld/plugins/dial_mobile/services.py [hunk 1]
```diff
@@ -865,6 +872,50 @@ def complete_pending_push_mobile(self, call_id: str) -> None:
...
+        )['value']
+        for contact in raw_contacts.split(','):
+            if not contact:
+                continue
+            mobility = self._ari.channels.getChannelVar(
+                channelId=channel_id, variable=f'PJSIP_CONTACT({contact},mobility)'
+            )['value']
+            if mobility == 'mobile':
+                return True
+        return False
```
[the remark this answers is not in the document]
Comment (fblackburn1) [reply wazo_calld/plugins/dial_mobile/services.py:928]: Valid observation, but it is a pre-existing gap, not a regression: the AOR-wide mobility scan is inherited verbatim from `has_a_registered_mobile_and_pending_push`, and on master the observable behavior in this scenario is identical (the state was always pruned before the decision ran, so no cancel push was sent either).

Also, a deskphone is a separate line/endpoint in Wazo, so its AOR has no mobile contact and takes the cancel path correctly. The remaining gap is the desktop/web app sharing the mobile's WebRTC line (same AOR): a proper fix is to match the answering channel to the exact contact the dialer originated it for, instead of scanning the AOR. Follow-up planned for that; keeping this PR scoped to the dead completion path.

File: integration_tests/suite/test_dial_mobile.py [hunk 2]
```diff
@@ -463,3 +464,69 @@ def push_notification_cancelled():
...
+                cancel_events.accumulate(),
+                has_item(has_entries(name='call_cancel_push_notification')),
+            )
+
+        until.assert_(
+            push_notification_cancelled,
+            timeout=5,
+            message=(
+                'push was not cancelled at join time; the push state was '
+                'probably pruned before being resolved'
```
Comment (DrPyser) [inline integration_tests/suite/test_dial_mobile.py]:
```suggestion
                'push was not cancelled at join time'
```
now that this hypothesis is resolved, a future failure of this test may be due to different causes.

**Review by DrPyser — DISMISSED** [review]
LGTM (trusting load tests)

**Review by DrPyser — APPROVED** [review]
LGTM, please merge
